lib/identity_cache/query_api.rb

```diff
@@ -483,13 +479,17 @@ def was_new_record? # :nodoc:
       !destroyed? && transaction_changed_attributes.has_key?(pk) && transaction_changed_attributes[pk].nil?
     end
 
-    def load_and_readonlyify(association_name)
-      record_or_records = send(association_name)
+    def readonly_record_copy(record)
+      record = record.dup
+      record.readonly!
+      record
+    end
 
-      if self.class.reflect_on_association(association_name).collection?
-        record_or_records.map { |p| p.dup.tap(&:readonly!) }
-      else
-        record_or_records.dup.tap(&:readonly!) if record_or_records
+    def readonly_copy(record_or_records)
+      if record_or_records.respond_to?(:to_ary)
+        record_or_records.map { |record| readonly_record_copy(record) }
+      elsif record_or_records
```

Comment:
yes, `record_or_records` is the result of loading an association, where a has_one association could return `nil`